sdk/samples/iot/paho_iot_provisioning_sample.c

```diff
@@ -323,6 +323,16 @@ static void handle_device_registration_status_message(
           "Hub Hostname:", register_response->registration_state.assigned_hub_hostname);
       IOT_SAMPLE_LOG_AZ_SPAN("Device Id:", register_response->registration_state.device_id);
       IOT_SAMPLE_LOG(" "); // Formatting
+
+      if (az_span_size(register_response->registration_state.payload) > 0)
+      {
+        IOT_SAMPLE_LOG_SUCCESS("Optional payload (Custom Allocation Policy) received:");
```

Comment:
might not want to tie this to "Custom Allocation Policy" because in the future they might allow azure functions that is not tied to Custom Allocation Policies